bundles/org.openhab.binding.bluetooth.airthings/src/main/java/org/openhab/binding/bluetooth/airthings/internal/AirthingsWavePlusHandler.java

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.airthings.internal;
+
+import static org.openhab.binding.bluetooth.airthings.internal.AirthingsBindingConstants.*;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.bluetooth.BeaconBluetoothHandler;
+import org.openhab.binding.bluetooth.BluetoothCharacteristic;
+import org.openhab.binding.bluetooth.BluetoothCompletionStatus;
+import org.openhab.binding.bluetooth.BluetoothDevice.ConnectionState;
+import org.openhab.binding.bluetooth.notification.BluetoothConnectionStatusNotification;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AirthingsWavePlusHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+@NonNullByDefault
+public class AirthingsWavePlusHandler extends BeaconBluetoothHandler {
+
+    private static final String DATA_UUID = "b42e2a68-ade7-11e4-89d3-123b93f75cba";
+
+    private final Logger logger = LoggerFactory.getLogger(AirthingsWavePlusHandler.class);
+    private final UUID uuid = UUID.fromString(DATA_UUID);
+
+    private volatile Boolean servicesResolved = false;
+    private Optional<AirthingsConfiguration> configuration = Optional.empty();
+    private @Nullable ScheduledFuture<?> scheduledTask;
+
+    public AirthingsWavePlusHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        configuration = Optional.of(getConfigAs(AirthingsConfiguration.class));
+        logger.debug("Using configuration: {}", configuration);
+    }
+
+    @Override
+    public void dispose() {
+        try {
+            super.dispose();
+        } finally {
+            if (scheduledTask != null) {
+                scheduledTask.cancel(true);
+                scheduledTask = null;
+            }
+            servicesResolved = false;
+        }
+    }
+
+    private void startScheduledTask() {
```

Comment:
why isn't this called during initialization?

Reply:
If device is not yet discovered (which is not happened during initialisation), first task execution will "fail". If user have configured poll intervall e.g. for an hour, next/first poll will happen rather late.

Reply:
You only start the scheduled task when the handler receives a connections state change to the `DISCOVERED` state. Which requires two things to happen:
1. That the device must switch to the DISCOVERED state, which only happens when the device is initially seen by an adapter. Also, almost all device immediately switch from DISCOVERED to DISCONNECTED, so simply checking if the device is in the DISCOVERED state will often not work.
2. That the handler must be listening to the device when it changes to its DISCOVERED state. This requires that the handler be initialized and running before the device is discovered by a bluetooth adapter. Which means that the handler will not work if the user tries to create the handler from an inbox discovery; If it is in the inbox, that means it was already discovered!

Not to mention that there is no guarantee on openhab startup that the `Airthings` handler will be started before the bluetooth device handlers. In fact, by virtue of the bluetooth handlers using the bluetooth adapters as a bridge, it is almost guaranteed that the adapters will start first. If the device is in range during that time your handler could very well miss the DISCOVERED event.

To summarize: make sure that your scheduled task lifecycle is tied to the handler and not dependent on device state.